src/components/TodoItem/TodoItem.tsx

```diff
@@ -0,0 +1,105 @@
+/* eslint-disable jsx-a11y/control-has-associated-label */
+import React, {
+  useContext,
+  useEffect,
+  useRef,
+  useState,
+} from 'react';
+import classNames from 'classnames';
+import { Todo } from '../../types/Todo';
+import { TodoContext } from '../../context/TodoContext';
+
+type Props = {
+  todo: Todo
+};
+
+export const TodoItem: React.FC<Props> = React.memo(({ todo }) => {
+  const { deleteTodo, updateTodo } = useContext(TodoContext);
+  const { title, id, completed } = todo;
+
+  const [newTitle, setNewTitle] = useState(title);
+  const [editing, setEditing] = useState(false);
```

Comment:
Rename all boolean variables according [tips](https://dev.to/michi/tips-on-naming-boolean-variables-cleaner-code-35ig)

```suggestion
  const [isEditing, setIsEditing] = useState(false);
```